lib/util/identifier.js

```diff
@@ -243,3 +243,19 @@ const _absolutify = (context, request) => {
 
 const absolutify = makeCacheable(_absolutify);
 exports.absolutify = absolutify;
+
+const PATH_QUERY_FRAGMENT_REGEXP = /^([^?#]*)(\?[^#]*)?(#.*)?$/;
+
+/**
+ * @param {string} str the path with query and fragment
+ * @returns {{ path: string, query: string, fragment: string }} parsed parts
+ */
+const parsePathQueryFragment = str => {
```

Comment:
maybe cache this.. for 1000 resources (!== requests) with average 30chars (assume all in 0000..007F - Basic Latin or
0080..00FF Latin-1 Supplement) don't know for sure I think this will be 2bytes size.. utf16?? `~= 60KB`

Reply:
Yep. Luckily it's in the correct file for caching...